doc/operation-and-maintenance/Rolling-upgrade.md

```diff
@@ -0,0 +1,85 @@
+## Rolling upgrade
+For all MongooseIM production deployments we recommend running multiple server nodes connected in a cluster behind a load-balancer.
+Rolling upgrade is a process of upgrading MongooseIM cluster, one node at a time.
+Make sure you have at least the number of nodes able to handle your traffic plus one before the rolling upgrade to guarantee the availability and minimise the downtime.
+Running different MongooseIM versions at the same time beyond the duration of the upgrade is not recommended and not supported.
+
+Rolling upgrade procedure is recommended over configuration reload which is not supported since version 4.1.
```

Comment:
It is a bit hard to imagine how the process looks like if only the configuration file is updated. Could we add a note that you only need to do steps 1, 3, 4, 6-8 and in step 3 what you do is actually updating the config file according to your needs? Alternatively, we could add notes directly to steps 2, 3 and 5.

Maybe a paragraph about the variant of this procedure for config update?

Now it might be confusing that the description does not say anything about updating the config file, so it might give an impression that this description is not relevant to updating the config.

Reply:
Good point! I'll have that changed.